mantidproject/mantid
# Add table view for group peak workspace

### Description of work
This pull request introduces a new table view for group peaks workspaces. The new group table view will display the same data as the table view for single peak workspaces but with additional columns for the index of the peak workspace in the group and the index of the spectrum in the peak workspace. Moreover, it has the same features as the single peak workspace table view like sorting, deleting, copying, and plotting. 

Fixes #36592. 


#### Further detail of work

The work is implemented without adding a new data structure for the group peak workspace as it is not needed and all of the table view operations can be done with the existing data structure. The new view has been implemented by extending the existing table display MVP classes with new presenters and models for the group workspace.

### To test:
Load and Group Workspaces:
- Load multiple peak workspaces and verify they appear in the workspace tree.
- Use the "GroupWorkspace" algorithm to group them.

Verify Group Data View:
- Right-click the group in ADS and select "Show Data."
- Verify the new table view has the same data of all the workspaces in the group.
- The table should include "Group Index" indicating workspace position in the group and "WS Index" showing the workspace index in the corresponding peak workspace.

Interact with the Table View:
- Test copying data (single/multiple rows, single/multiple columns) to the clipboard and paste it in another file.
- Test deleting rows (single/multiple) and confirm successful removal.
- Sort data by column (ascending/descending) and ensure correct sorting.
- Select "Statistics on Columns" and verify the output includes tables with different statistics (e.g., Mean, Std Dev, Min/Max) for each column in each peak workspace
- Select columns for 'X', 'Y', and 'Y Error'. Choose various plot types and check the data is visualized correctly.
- Test hiding/unhiding columns 





---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
